build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/YamlRestCompatTestPlugin.java

```diff
@@ -142,7 +143,11 @@ public void apply(Project project) {
                 task.getOutputResourceDir().set(projectLayout.getBuildDirectory().dir(compatTestsDir.resolve("original").toString()));
                 task.setCoreConfigToFileTree(
                     config -> fileOperations.fileTree(
-                        config.getSingleFile().toPath().resolve(RELATIVE_REST_API_RESOURCES).resolve(RELATIVE_TEST_PATH)
+                        config.getSingleFile()
+                            .toPath()
+                            .resolve(RELATIVE_REST_CORE)
+                            .resolve(RELATIVE_REST_PROJECT_RESOURCES)
+                            .resolve(RELATIVE_TEST_PATH)
```

Comment:
The old path didn't exist. I could have merged the paths some more but I don't feel particularly confident working on this code. I am a little worried what'll happen when I "fix" this. Will we start running tests we hadn't been running before? CI will tell me if any of them fail at least....

Reply:
Ah, so this was just _wrong_ before I assume?

Reply:
I think so! The path didn't exist and nothing was ever copied.

Reply:
Indeed this is a bug. Fortunately it only impacts copying tests from core (`includeCore`) for modules that have the yaml-rest-compat-test plugin. It seems that this only impacts this aggs module. 

However, while reviewing i see that we have 2 projects (enrich, watcher with security) attempting to copy some tests from xpack (`includeXpack`) that are not running the compat tests as they should be since the path to the x-pack tests is also wrong. I will followup on that in a different PR.

Reply:
https://github.com/elastic/elasticsearch/pull/90925 to fix the x-pack path, and https://github.com/elastic/elasticsearch/issues/90929 to be a bit smarter here.